src/Microsoft.IdentityModel.Abstractions/ITelemetryClient.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.IdentityModel.Abstractions
+{
+    /// <summary>
+    /// Interface for Telemetry tracking.
+    /// </summary>
+    public interface ITelemetryClient
+    {
+        /// <summary>
+        /// Gets or sets the application or client ID that telemetry is being sent for.
+        /// </summary>
+        string ClientId { get; set; }
+
+        /// <summary>
+        /// Perform any necessary bootstrapping for the telemetry client.
+        /// </summary>
+        /// <remarks>
+        /// The expectation is that this should only be called once for the lifetime of the object however the
+        /// implementation should be idempotent.
+        /// </remarks>
+        void Initialize();
+
+        /// <summary>
+        /// Checks to see if telemetry is enabled all up.
+        /// </summary>
+        /// <returns>
+        /// Returns <see langword="true"/> if telemetry should be sent; <see langword="false"/> otherwise.
+        /// </returns>
+        /// <remarks>
+        /// This check should be used to gate any resource intensive operations to generate telemetry as well.
+        /// </remarks>
+        bool IsEnabled();
+
+        /// <summary>
+        /// Checks to see if telemetry is enabled for the named event.
+        /// </summary>
+        /// <param name="eventName">Name of the event to check.</param>
+        /// <returns>
+        /// Returns <see langword="true"/> if telemetry should be sent for <paramref name="eventName"/>;
+        /// <see langword="false"/> otherwise.
+        /// </returns>
+        /// <remarks>
+        /// This check should be used to gate any resource intensive operations to generate telemetry as well.
+        /// </remarks>
+        bool IsEnabled(string eventName);
+
+        /// <summary>
+        /// Tracks an instance of a named event.
+        /// </summary>
+        /// <param name="eventDetails">Details of the event to track.</param>
+        void TrackEvent(
```

Comment:
Perhaps this should expose a TrackEventAsync As well or paerhaps there should be an ITelemetryClientAsync instead

Reply:
Can expose a different interface for that later if it becomes required.